src/GrasshopperTeklaDrawingLink/Components/Text/ArrowAttributesComponent.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Grasshopper.Kernel;
+
+using GTDrawingLink.Tools;
+using GTDrawingLink.Types;
+
+using System.Collections.Generic;
+
+using Tekla.Structures.Drawing;
+
+namespace GTDrawingLink.Components.Text
+{
+    public class ArrowAttributesComponent : TeklaComponentBase
+    {
+        public ArrowAttributesComponent()
+            : base(ComponentInfos.ArrowAttributesComponent)
+        {
+        }
+        public override GH_Exposure Exposure => GH_Exposure.primary;
+        protected override void RegisterInputParams(GH_InputParamManager pManager)
+        {
+            SetParametersAsOptional(pManager, new List<int> {
+            pManager.AddParameter(new EnumParam<ArrowheadTypes>(ParamInfos.ArrowType, GH_ParamAccess.item))
+            });
+            AddNumberParameter(pManager, ParamInfos.Width, GH_ParamAccess.item, true);
+            AddNumberParameter(pManager, ParamInfos.Height, GH_ParamAccess.item, true);
+        }
+
+        protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+        {
+            pManager.AddParameter(new ArrowAttributesParam(ParamInfos.ArrowAttribute, GH_ParamAccess.item));
+        }
+
+        protected override void SolveInstance(IGH_DataAccess DA)
+        {
+            var arrowheadAttributes = new ArrowheadAttributes();
+            object arrowType = null;
+            double height = 0.0;
+            double width = 0.0;
+            DA.GetData(ParamInfos.ArrowType.Name, ref arrowType);
+            DA.GetData(ParamInfos.Height.Name, ref height);
+            DA.GetData(ParamInfos.Width.Name, ref width);
+
+
+            height = (height == 0) ? new ArrowheadAttributes().Height : height;
```

Comment:
In line 35 we create instance of ArrowheadAttributes. By default it sets height as 2.0 and width as 1.0.

Then, in 44 and 45 we check default value of inputs (heigh, width) and if user has not specified those arguments, then we are once again creating ArrowheadAttributes object to get its default properties.

Simplier version would be just to wrap exisitng lines 51 and 52 with checks like:
`
if (width > 0)
   arrowheadAttributes.Width = width;
`